### DISCUSSION TITLE
Access Knowledgebase via api?

### DISCUSSION THREAD

**Discussion by Steff81** [open-webui/open-webui#7550] [General]
I have created an own model in openwebui that has attached a knowledgebase. The prompt says it should look up data always first in the knowledgebase before answering the question. 

In the openwebui interface all of this works perfectly fine as it should. But via API the model is totally ignoring the knowledgebase. 
I have seen that there is a difference on the openwebui console if i made a request via the openwebui interface in the browser or via api. 

Only when i ask the model via the browser interface an endpoint api/v1/memories is called. 
This call is missing when I use the api endpoints even the knowledebase is attached to the model.

So is this behavior intended? Or is this a bug or do I do something totally wrong. 
What i also tried is to use the files parameter in the POST endpoint of /api/chat but this leads to an exception in the openwebui console.

**Comment by Steff81** [comment]
Found the solution. Works. 
Nevermind. 
For everyone facing the same problem, you can add the collection with the id (not the name) to the request like below.

```
		$post_data = [
			"model"       => $this->Model,
			// "max_tokens"  => $this->MaxTokens, // not used in ollama
		  "options" => [
		    "temperature" => $this->Temperature
		  ],
			'stream'      => false,
			"messages"    => [
				[	"role" => "system", "content" => '', ],
				[	"role" => "user",   "content" => $this->Prompt, ]
			],
			"files" => [
        		["type" =>  "collection", "id" =>  "d2fa63ee-7536-48f2-8773-b3e0f37c08e2"]
        	],
		];
```

**Reply by VicentePina7210** [reply]
thank you for posting this, the documentation fails to explain some of these basic concepts.
